sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -64,17 +62,6 @@ class QueryExecution(
   // TODO: Move the planner an optimizer into here from SessionState.
   protected def planner = sparkSession.sessionState.planner
 
-  // The CTE map for the planner shared by the main query and all subqueries.
-  private val cteMap = mutable.HashMap.empty[Long, CTERelationDef]
```

Comment:
This move (from `QueryExecutor` to the other location) could be the root cause of the UT failure in GitHub Action.

As I posted on https://github.com/apache/spark/pull/36815 , this PR seems to generate the golden answer files with the following correctly.
```
SPARK_GENERATE_GOLDEN_FILES=1 build/sbt "sql/testOnly *PlanStabilitySuite"
```

However, unfortunately, we are experiencing GitHub Action failures. In addition, the result is also different from the one when we run individual query.
```
SPARK_GENERATE_GOLDEN_FILES=1 build/sbt "sql/testOnly *PlanStabilitySuite -- -z (tpcds-v1.4/q4)"
```

Given that, this PR seems to introduce indeterministic logic in terms of expression IDs.

Reply:
Could you take a look together with me, @maryannxue and @cloud-fan ?

If there is no easy solution, I'd like to recommend to revert this from `branch-3.2` first because it has been broken for 2 months already. How do you want to proceed this?

Reply:
Here is the result generation status.
```
$ SPARK_GENERATE_GOLDEN_FILES=1 build/sbt "sql/testOnly *PlanStabilitySuite" &> /dev/null
$ git status
On branch branch-3.2
Your branch is up to date with 'apache/branch-3.2'.

nothing to commit, working tree clean

$ SPARK_GENERATE_GOLDEN_FILES=1 build/sbt "sql/testOnly *PlanStabilitySuite -- -z (tpcds-v1.4/q4)" &> /dev/null
$ SPARK_GENERATE_GOLDEN_FILES=1 build/sbt "sql/testOnly *PlanStabilitySuite -- -z (tpcds-v1.4/q5)" &> /dev/null
$ git diff --stat
 sql/core/src/test/resources/tpcds-plan-stability/approved-plans-v1_4/q4/explain.txt | 238 ++++++++++++++++++++++++++++++++++++++++++++++++++++----------------------------------------------------
 sql/core/src/test/resources/tpcds-plan-stability/approved-plans-v1_4/q5/explain.txt | 212 ++++++++++++++++++++++++++++++++++++++++++++++----------------------------------------------
 2 files changed, 225 insertions(+), 225 deletions(-)
```

Reply:
It turns out to be a bug in the plan stability test suite.

The test suite normalizes the expr IDs, by using regex `"#\\d+L?".r` to math the explain string. However, The exchange node has a special string arg `id=#...`: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/Exchange.scala#L40

The regex can't distinguish between expr ID and exchange plan id, and may normalize the plan wrongly.

I'll try to fix it tomorrow.

Reply:
It looks like a general bug on all branches. In that case, do you know why only `branch-3.2` is so flaky like this?

Reply:
if the plans are different between 3.2 and 3.3, then the plan ids are different and we may not trigger the bug.